src/controller/GameManager.java

```diff
@@ -45,23 +44,16 @@ private void choosePlayerTypes() {
     }
 
     private void determineWinner() {
-        final boolean whiteHasNoPieces = this.gameBoard.getNumberOfWhitePieces() == 0;
-        final boolean whiteHasNoMoves = MoveGenerator
-                .getAllPossibleMoves(this.gameBoard, PieceColor.WHITE).isEmpty();
-        final boolean blackHasNoPieces = this.gameBoard.getNumberOfBlackPieces() == 0;
-        final boolean blackHasNoMoves = MoveGenerator
-                .getAllPossibleMoves(this.gameBoard, PieceColor.BLACK).isEmpty();
+        final PieceColor winner = this.gameBoard.determineWinner();
 
-        final boolean whiteWins = blackHasNoPieces || blackHasNoMoves;
-        final boolean blackWins = whiteHasNoPieces || whiteHasNoMoves;
-
-        if (whiteWins) {
+        if (winner == null) {
```

Comment:
Small change, but if you're going to put the error case check in here, it's probably better to make it:

```
if (winner == PieceColor.WHITE) {
    // white wins
} else if (winner == PieceColor.BLACK) {
    // black wins
} else {
    // invalid ending state
}